server/routes/productRoute.js

```diff
@@ -3,32 +3,71 @@ const router = express.Router();
 const productData = require("../data/products");
 const session = require("express-session");
 
+// router.get("/", async (req, res) => {
+//   try {
+//     let prodList = await productData.getAllProducts();
+//     res.status(200).render("homePage/homePage", {
+//       products: prodList,
+//     });
+//   } catch (e) {
+//     console.log(e);
+//     res.status(404).send(e);
+//   }
+// });
+
 router.post("/search", async (req, res) => {
-    const body = req.body;
-    let option = document.getElementById("searchSelect");
-    let searchTerm = body.search;
-    if (!searchTerm || searchTerm.trim() == 0) {
-        return "Error: Search term blank";
-    } else {
-        searchTerm = searchTerm.toLowerCase();
-        if (option.value === "By Name") {
-            try {
-                let search_List = await products.getProductByProductName(
-                    body.search
-                );
-                res.status(200).render("homePage/homePage", {
-                    products: search_List,
-                });
-            } catch (e) {
-                return res.status(500).message(e.message);
-            }
-        }
-        if (option.value === "By Tag") {
-            try {
-                let search_List = await products.getProductbyTag(body.search);
-            } catch (e) {
-                return res.status(500).send();
-            }
-        }
+  const body = req.body;
+  console.log("body", body);
+
+  //console.log(option);
+  let searchTerm = body.search;
+  console.log(searchTerm);
+  if (!searchTerm || searchTerm.trim() == 0) {
```

Comment:
I think it should be searchterm.trim().length

Reply:
yes Sorry I think I missed this.